v2/api/apimanagement/v1api20220801/storage/structure.txt

```diff
@@ -155,6 +155,80 @@ github.com/Azure/azure-service-operator/v2/api/apimanagement/v1api20220801/stora
 │       ├── VersionHeaderName: *string
 │       ├── VersionQueryName: *string
 │       └── VersioningScheme: *string
+├── AuthorizationProvider: Resource
+│   ├── Owner: apimanagement/v1api20220801.Service
+│   ├── Spec: Object (7 properties)
+│   │   ├── AzureName: string
+│   │   ├── DisplayName: *string
+│   │   ├── IdentityProvider: *string
+│   │   ├── Oauth2: *Object (3 properties)
+│   │   │   ├── GrantTypes: *Object (3 properties)
+│   │   │   │   ├── AuthorizationCode: *genruntime.SecretMapReference
+│   │   │   │   ├── ClientCredentials: *genruntime.SecretMapReference
+│   │   │   │   └── PropertyBag: genruntime.PropertyBag
+│   │   │   ├── PropertyBag: genruntime.PropertyBag
+│   │   │   └── RedirectUrl: *string
+│   │   ├── OriginalVersion: string
+│   │   ├── Owner: *genruntime.KnownResourceReference
+│   │   └── PropertyBag: genruntime.PropertyBag
+│   └── Status: Object (8 properties)
+│       ├── Conditions: conditions.Condition[]
+│       ├── DisplayName: *string
+│       ├── Id: *string
+│       ├── IdentityProvider: *string
+│       ├── Name: *string
+│       ├── Oauth2: *Object (3 properties)
+│       │   ├── GrantTypes: *Object (3 properties)
+│       │   │   ├── AuthorizationCode: map[string]string
+│       │   │   ├── ClientCredentials: map[string]string
```

Comment:
Why is GrantTypes.* above *genruntime.SecretMapReference, but here a map of string?

Reply:
This is the `Status` version of the object, capturing the information returned from ARM when ASO does a `GET`.
A few lines above (on line #167 as I write) is the `Spec` version, which details where to pull the secret before ASO sends the resource to ARM with a PUT.